kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/V3ResourcesModule.java

```diff
@@ -30,5 +39,38 @@ protected void configure() {
     bindAsContract(ChunkedOutputFactory.class);
     bindAsContract(StreamingResponseFactory.class);
     bind(Clock.systemUTC()).to(Clock.class);
+    bindFactory(ProduceResponseExecutorServiceFactory.class)
+        .to(ExecutorService.class)
```

Comment:
I still think it's better to not have this act as a generic `ExecutorService` factory.

Reply:
@ehumber can you add a `@ProduceResponseThreadPool` annotation to this executor?

Reply:
@dimitarndimitrov is just naming it to have ProduceResponse in the title sufficient, or were you thinking of something additional?  In which case, can you explain in a little more detail please.

Reply:
@rigelbm I think I've done that in the commit I'm about to push, but let me know if it's not what you mean.

Reply:
I think the problem is not with the name of the factory type, but with the registered type provided via that factory, which currently still seems to be the generic `ExecutorService` type. My inelegant suggestion would have been to introduce an `ExecutorService` subtype which to provide with this factory, and inject that subtype where we need it. I think Rigel is saying that this can also be done by simply annotating the injection point and the binding here with a new annotation specifically for the executor "flavor". Not sure how exactly does this need to look, but it sounds preferable to me.

Reply:
```
bindFactory(ProduceResponseExecutorServiceFactory.class)
    .qualifiedBy(new ProduceResponseThreadPoolImpl())
    .to(ExecutorService.class)
    .in(Singleton.class);

//...

@Qualifier
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER})
public @interface ProduceResponseThreadPool {}

private static final class ProduceResponseThreadPoolImpl
    extends AnnotationLiteral<ProduceResponseThreadPool>
    implements ProduceResponseThreadPool {}

// ...

private static final class ProduceResponseExecutorServiceFactory
    implements Factory<ExecutorService> {
    // ....
}
```